Sources/SotoCore/Credential/STSAssumeRole.swift

```diff
@@ -127,7 +127,10 @@ struct STSAssumeRoleCredentialProvider: CredentialProviderWithClient {
     func getCredential(on eventLoop: EventLoop, logger: Logger) -> EventLoopFuture<Credential> {
         self.assumeRole(self.request, logger: logger, on: eventLoop)
             .flatMapThrowing { response in
-                guard let credentials = response.credentials else { throw CredentialProviderError.noProvider }
+                guard let credentials = response.credentials else {
+                    throw CredentialProviderError.noProvider
+                }
+                _ = self.shutdown(on: eventLoop)
```

Comment:
Why are you shutting down the credential provider here? This is done when the parent AWSClient is shutdown. If you shut it down now, 1) the client will complain when you shut it down and 2) when you need to refresh credentials the provider will have already been shutdown

Reply:
This was raising an error when running end-to-end tests that used `STSAssumeRole`, as the client was not being shutdown after successfully retrieving temporary credentials from STS. 

There might be a better way to do it, will look into it.

Reply:
Think I know what it is `CredentialProviderSelector.shutdown` should be calling `shutdown` on the provider it holds
```swift 
return self.startupPromise.futureResult.flatMap { provider in provider.shutdown()}.hop(to: eventLoop)
```
or something like that

Reply:
If you rebase with main the `CredentialProviderSelector.shutdown` will be fixed